### PR TITLE
Pallet Assets: Create new asset classes from genesis config

### PR DESCRIPTION
**Pull request by vgeddes**
This change allows chain developers to create new asset classes via genesis config. This is useful for asset classes controlled by the by the root account of a parachain.

Currently the assets pallet requires that stakeholders create a new asset class via a dispatchable (presumably invoked via an extrinsic). This limits the utility of the assets pallets. Our parachain would like to create new asset classes via a loosely coupled trait,  and/or via genesis config. This PR addresses the latter.

If you think this is the right approach, I can go ahead and create companion PRs for Statemint, etc in the Cumulus repo

Polkadot address: 13qBjpFmDbf5vyumEcqpUji5V2iVzsZb6oN7d1qH7i2ZRHep

### CODE REVIEW

**Comment by shawntabrizi** [comment]
This seems fine, but you dont want to give any genesis balances?

I would expect a vec of `Vec<(AccountId, Balance)>` for giving users balances at the start.

**Comment by vgeddes** [comment]
I agree, I'll add that too. Also for asset metadata.

File: frame/assets/src/lib.rs [hunk 1]
```diff
@@ -275,6 +290,89 @@ pub mod pallet {
...
+						is_frozen: false,
+					},
+				);
+			}
+
+			for (id, name, symbol, decimals) in &self.metadata {
+				assert!(Asset::<T, I>::contains_key(id), "Asset does not exist");
+
+				let bounded_name: BoundedVec<u8, T::StringLimit> =
+					name.clone().try_into().expect("string is too long");
```
Comment (shawntabrizi) [inline frame/assets/src/lib.rs:343]:
```suggestion
					name.clone().try_into().expect("asset name is too long");
```

File: frame/assets/src/lib.rs [hunk 2]
```diff
@@ -275,6 +290,89 @@ pub mod pallet {
...
+				let bounded_symbol: BoundedVec<u8, T::StringLimit> =
+					symbol.clone().try_into().expect("string is too long");
```
Comment (shawntabrizi) [inline frame/assets/src/lib.rs:345]:
```suggestion
					symbol.clone().try_into().expect("asset symbol is too long");
```

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
@vgeddes please merge master.

**Comment by vgeddes** [comment]
Awesome, thanks for the tip! I've merged in master and pasted my address in the PR description, as well as below:

Polkadot address: 13qBjpFmDbf5vyumEcqpUji5V2iVzsZb6oN7d1qH7i2ZRHep

**Review by emostov — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**